integration/networking/port_mapping_linux_test.go

```diff
@@ -362,3 +362,44 @@ func TestAccessPublishedPortFromRemoteHost(t *testing.T) {
 		}
 	}
 }
+
+func TestRestartUserlandProxyUnder2MSL(t *testing.T) {
+	skip.If(t, testEnv.IsRootless())
+
+	ctx := setupTest(t)
+
+	d := daemon.New(t)
+	d.StartWithBusybox(ctx, t)
+	defer d.Stop(t)
+
+	c := d.NewClientT(t)
+	defer c.Close()
+
+	network.CreateNoError(ctx, t, c, "nat-time-wait",
+		network.WithDriver("bridge"),
+		network.WithOption("com.docker.network.bridge.name", "nat-time-wait"))
+	defer network.RemoveNoError(ctx, t, c, "nat-time-wait")
+
+	serverID := container.Run(ctx, t, c,
+		container.WithName(sanitizeCtrName(t.Name()+"-server")),
+		container.WithExposedPorts("80/tcp"),
+		container.WithPortMap(nat.PortMap{"80/tcp": {{HostPort: "1780"}}}),
+		container.WithCmd("httpd", "-f"),
+		container.WithNetworkMode("nat-time-wait"))
```

Comment:
Could make a `[]func(*TestContainerConfig)` of these options and use it in both `Run` calls - just to make it more obvious that the networks are the same, and in case something needs to change one day?